highfidelity/hifi
# Update model render items on texture load and fix no haze issue

https://highfidelity.manuscript.com/f/cases/11053/Transparent-materials-render-opaque

When a model overlay/entity is created, its URL changes, or its texture URLs change, we must wait until the textures load and then trigger a render update to properly mark the render items as opaque/transparent.

Also, in cases where there was no haze stage, we would not bind anything, so the buffer would be full of junk, instead of the default "off" values.

Test plan:
- Use teleport.  The model overlay at the end should render correctly.
- Find a model with transparent textures (there are plenty in the Volcano Lair content set - I used the light fixtures above the submarine).  It should render correctly.
- Reload content several times.  The above steps should still work after each reload.
- Go to an empty location.  You should not see any haze.
- From the empty location, create a model entity with transparent parts and create a zone around it with haze set to inherit.  Inside the zone, you should not see any haze and the object should render correctly.
- Change the zone's haze property to on.  You should see haze.
- Change it to off.  You should not see haze.
